09-Michael-A-Anderson/wk01/5-fri/MTA/js/script.js

```diff
@@ -0,0 +1,125 @@
+    stationFactory('6','28th'),
+    stationFactory('6','23rd'),
+    stationFactory('6','Union Square'),
+    stationFactory('6','Astor')
+]
+
+tripadvisor('6','Grand Central','6','23rd');
+tripadvisor('6','23rd','6','Grand Central');
+tripadvisor('6','23rd','L','3rd');
+
+function planTrip()
+{
+    onLine = prompt("What line are you getting on")
+    onStop = prompt("What stop on the line are you getting on")
+    offLine = prompt("What line are you getting off")
+    offStop = prompt("What stop on the line are you getting off")
+    tripadvisor(onLine,onStop,offLine,offStop)
+}
+
+function tripadvisor(onLine,onStop,offLine,offStop){
+    const on = findLine(onLine,onStop) // returns array with line and array index for stop
+    const off = findLine(offLine,offStop)// returns array with line and array index for stop    
+    if(onLine===offLine){
+        const stops = legBuilder(on, off) //returns array of stops
+        displayStops(stops);
+    }else{
+        const legOneOff = findLine(onLine,'Union Square') // returns array with line and array index for stop
+        const legTwoOn = findLine(offLine,'Union Square') // returns array with line and array index for stop
+        // console.log(legOneOff)
+        const stops1 = legBuilder(on, legOneOff);//returns array of stops
+        stops1[stops1.length-1]+=` ** change to ${offLine} line here**`; //directions to change and what line to change to
+        const stops2 = legBuilder(legTwoOn, off);//returns array of stops
+        const stops = stops1.concat(stops2) // joins arrays
+        displayStops(stops);
+    }
+}
+
+function findLine(line,stop){ // selects the appropriate array
+    let station
+    switch (line){
+        case 'N': 
+            station = findStop(N,stop);
+            break;
+        case 'L':
+            station = findStop(L,stop);
+            break;
+        case '6':
+            station = findStop(six,stop);
+            break;
+    }
+    return station
+}
+
+//findLine('6','33rd');
+
+function findStop(line,stop){
+    for (let i=0;i<line.length;i++){
+        if(line[i].name===stop){
+            const station = [line, i];
+            // console.log('stop found')
+            return station;
+        }
+    }
+}
+
+function legBuilder(on,off){ // makes info more easily accessible and chooses a direction
+    const arr = on[0];
+    const start = on[1];
+    const finish = off[1];
+    // console.log(arr,start,finish)
+    if(on>off){
+        stops = backwards(arr,start,finish);
+    }else{
+        stops = forwards(arr,start,finish);
+    }
+    return stops;
+}
+
+function backwards(arr,on,off){ //loops through the line from point to point
+    let stops = []
+    for( let i = on-1;i>=off;i--){
+        stops.push(arr[i].name);
+    }
+    return stops;
+}
+
+function forwards(arr,on,off){ //loops through the line from point to point
+    let stops = []
+    for( let i = on+1;i<=off;i++){
+        stops.push(arr[i].name);
+    }
+    return stops;
+}
```

Comment:
There's a decent amount of repetition between these functions, if you wanted to refactor there might be creative ways to DRY this section up.

Reply:
On Sunday I messed around with the reverse() method to try and achieve this but it completely messed up my index's and needed to be factored into the solution much earlier.